SigScan.cpp

```diff
@@ -258,13 +258,9 @@ static int callback(struct dl_phdr_info *info, size_t size, void *data) {
 
             programBaseAddress = (uintptr_t) (info->dlpi_addr + t_phdr.p_vaddr);
             
-            size_t seg_size = 0;
             // get segment size
-            while(seg_size < t_phdr.p_filesz) {
```

Comment:
Okay @THETASOLA so I think this is because in the ELF spec there's multiple segments, so I was trying to go through every segment I was looking at in the file to find the total-end scanning range.

However; what it probably needs to be overall (if we're matching things in multiple segments, which we're probably not but when I was implementing this I was learning about ELF) is several different scanner ranges of starts & stops if we wanted to match across the different executable sections but that's a much bigger change to ZHL

Reply:
> I was looking at in the file to find the total-end scanning range

The previous code would also assign to segmentLength just with a wrong value so it wouldn't do that anyway and I think the exact same issue would occur.